src/django_components/__init__.py

```diff
@@ -21,20 +21,39 @@ def autodiscover() -> None:
         autodiscover_modules("components")
 
         # Autodetect a <component>.py file in a components dir
-        component_filepaths = search(search_glob="**/*.py")
+        component_filepaths = search(search_glob="**/*.py").matched_files
+        logger.debug(f"Autodiscover found {len(component_filepaths)} files in component directories.")
+
         for path in component_filepaths:
-            import_file(path)
+            # This imports the file and runs it's code. So if the file defines any
+            # django components, they will be registered.
+            module_name = _filepath_to_python_module(path)
+            logger.debug(f'Importing module "{module_name}" (derived from path "{path}")')
+            importlib.import_module(module_name)
 
     for path_lib in app_settings.LIBRARIES:
         importlib.import_module(path_lib)
 
 
-def import_file(path: Union[str, Path]) -> None:
-    MODULE_PATH = path
-    MODULE_NAME = Path(path).stem
-    spec = importlib.util.spec_from_file_location(MODULE_NAME, MODULE_PATH)
-    if spec is None:
-        raise ValueError(f"Cannot import file '{path}' - invalid path")
-    module = importlib.util.module_from_spec(spec)
-    sys.modules[spec.name] = module
-    spec.loader.exec_module(module)  # type: ignore
+def _filepath_to_python_module(file_path: Path) -> str:
+    """
+    Derive python import path from the filesystem path.
+
+    Example:
+    - If project root is `/path/to/project`
+    - And file_path is `/path/to/project/app/components/mycomp.py`
+    - Then the path relative to project root is `app/components/mycomp.py`
+    - Which we then turn into python import path `app.components.mycomp`
+    """
+    # Get the root dir, see https://stackoverflow.com/a/16413955/9788634
+    project_root = os.path.abspath(os.path.dirname(__name__))
```

Comment:
In a couple of places in the project we allow the user to set the BASE_DIR variable. Is this what you are trying to find here? Maybe we should use BASE_DIR if it's defined and fall back to this otherwise? https://github.com/search?q=repo%3AEmilStenstrom%2Fdjango-components%20BASE_DIR&type=code

Reply:
Yes, that's it :) Initially I avoided using BASE_DIR, because I was unfamiliar with it, and when I looked it up, it looked  like it was not a Django setting per se, but rather a convention, so I wasn't sure how widely it is used. Your suggestion with using BASE_DIR and falling back to the default above sounds like the best approach.

Reply:
Updated in https://github.com/EmilStenstrom/django-components/pull/435/commits/9104a439445ed074037049627bb8cd51e978d959 to use BASE_DIR